### DISCUSSION TITLE
Unit testing with net8.0-windows* framework

### DISCUSSION THREAD

**Discussion by jsiemensLatitudegeo** [dotnet/maui#18411] [General]
If I run this code:

    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethod1()
        {
            var label = new Label() { Text = "abc" };
        }
    }

Using:

    <TargetFramework>net8.0-windows10.0.19041.0</TargetFramework>
    <UseMaui>true</UseMaui>

Then I get: 
Message: 
Test method TestProject2.UnitTest1.TestMethod1 threw exception: 
System.TypeInitializationException: The type initializer for 'Microsoft.Maui.Controls.VisualElement' threw an exception. ---> System.TypeInitializationException: The type initializer for 'WinRT.ActivationFactory`1' threw an exception. ---> System.Runtime.InteropServices.COMException: Class not registered (0x80040154 (REGDB_E_CLASSNOTREG))

But if I run it with <TargetFramework>net8.0</TargetFramework> then it works fine. Because of a package dependency I cannot use 'net8.0' (platform neutral framework) - I must use a device framework.

Is there a way to get this code to work? Effectively, I don't want it to go and try to create the VisualElement. I just want it to do the platform neutral thing that it does with net8.0.

**Comment by jonathanMelly** [comment]
Hello I had a similar issue and the only way I found to make it work is to use a separate maui lib project and put all the code there that can be tested... I only validated that with my viewmodels classes but assume it should work for UI classes too...

**Comment by glowysourworm** [comment]
There's no Unit Test framework for .net v8.0-windows ?!?!?!  I'm getting a version error; and can't find any support! Please tell me that's not true....
